src/codegen/c.zig

```diff
@@ -2517,9 +2562,12 @@ fn airUnwrapErrUnionPay(f: *Function, inst: Air.Inst.Index) !CValue {
         return CValue.none;
     }
 
+    // cannot reliably determine from the return type whether we need to do the addrof ourselves
+    // e.g. *(E!T) -> *T   !=   E!*T -> *T
+    const inst_is_ptr = f.air.instructions.items(.tag)[inst] == .unwrap_errunion_payload_ptr;
```

Comment:
suggestion: pass `inst_is_ptr` as a parameter to this function rather than doing this check again. Better yet, pass `maybe_addrof` instead.